### ISSUE TITLE
@click events don't trigger in x-html rendered nested elements

### ISSUE THREAD

**Issue by jreviews** [alpinejs/alpine#86]
The @click event in the code below doesn't trigger after `this.tasks` **changes**. It does work when the `renderList` function runs on component initialization.

```
renderList() { 
	return this.tasks.map(task => {
	    return `
			<div>
			 	<span x-text="'${task}'"></span> 
			 	<button @click="removeTask('${task}')">Remove</button>
			</div>
		`
	  }).join('')
}
```

However, when adding the @click event to the parent 'div', then it works:

```
renderList() { 
	return this.tasks.map(task => {
	    return `
			<div @click="removeTask('${task}')">
			 	<span x-text="'${task}'"></span> 
			 	<button>Remove</button>
			</div>
		`
	  }).join('')
}
```

**Comment by jreviews** [comment]
I created an example here https://codepen.io/jreviews/pen/oNgPgXG

In the first list, the "remove" button stops working whenever `renderList` runs after the initial component setup. In the second list, there's no problem,

**Comment by jreviews** [comment]
This was fixed in 1.9.0
